kubernetes/VPASettings.dhall

```diff
@@ -0,0 +1,144 @@
+let imports = ../imports.dhall
+
+let Kubernetes = imports.Kubernetes
+
+let KubeUtils = imports.KubeUtils
+
+let Resources =
+      let Resources =
+            { limits :
+                { cpu : KubeUtils.Resources.Compute.Types
+                , memory : KubeUtils.Resources.Memory.Types
+                }
+            , requests :
+                { cpu : KubeUtils.Resources.Compute.Types
+                , memory : KubeUtils.Resources.Memory.Types
+                }
+            }
+
+      in  { Type = Resources
+          , default = {=}
+          , render.kubernetes-resource-requirements =
+                λ(resources : Resources)
+              → Kubernetes.ResourceRequirements::{
+                , limits = Some
+                    ( toMap
+                        { cpu =
+                            KubeUtils.Resources.Compute.render
+                              resources.limits.cpu
+                        , memory =
+                            KubeUtils.Resources.Memory.render
+                              resources.limits.memory
+                        }
+                    )
+                , requests = Some
+                    ( toMap
+                        { cpu =
+                            KubeUtils.Resources.Compute.render
+                              resources.requests.cpu
+                        , memory =
+                            KubeUtils.Resources.Memory.render
+                              resources.requests.memory
+                        }
+                    )
+                }
+          }
+
+let AdmissionController =
+      let TLS =
+            { Type =
+                { certificate-authority-private-key : Text
+                , certificate-authority-public-certificate : Text
+                , server-private-key : Text
+                , server-public-certificate : Text
+                }
+            , default = {=}
+            }
+
+      in  { Type =
+              { enabled : Bool
+              , tls : TLS.Type
+              , image : KubeUtils.Image.Types
+              , resources : Resources.Type
+              }
+          , default =
+              { enabled = True
+              , image =
+                  KubeUtils.Image.Types.Tag
+                    { name = "k8s.gcr.io/vpa-admission-controller"
+                    , tag = "0.6.1"
+                    }
+              , resources = Resources::{
+                , limits =
+                    { cpu = KubeUtils.Resources.Compute.Types.Milicpu 200
+                    , memory = KubeUtils.Resources.Memory.Types.MB 500
+                    }
+                , requests =
+                    { cpu = KubeUtils.Resources.Compute.Types.Milicpu 50
+                    , memory = KubeUtils.Resources.Memory.Types.MB 200
```

Comment:
should we export it to somewhere ?

Reply:
if we do `kubernetes/autoscaler/vertical-pod-autoscaler/settings.dhall` then maybe also have a `kubernetes/autoscaler/vertical-pod-autoscaler/core.dhall` that has the bits and pieces in common